src/wallet/wallet.cpp

```diff
@@ -289,6 +289,27 @@ void CWallet::SetBestChain(const CBlockLocator& loc)
 {
     CWalletDB walletdb(strWalletFile);
     walletdb.WriteBestBlock(loc);
+
+    UpdateConflicted();
+}
+
+void CWallet::UpdatedBlockTip(const CBlockIndex *pindex)
+{
+    UpdateConflicted();
+}
+
+void CWallet::UpdateConflicted()
+{
+    //check for conflicted transactions
+    LOCK2(cs_main, cs_wallet);
+    for (map<uint256, CWalletTx>::iterator it = mapWallet.begin(); it != mapWallet.end(); ++it)
+    {
+        CWalletTx* wtx = &((*it).second);
+
+        // update the conflicted cache flag
+        if (wtx->GetDepthInMainChain() <= 0)
```

Comment:
This is recursive logic. You're computing the value of the conflicted flag based on the output of GetDepthInMainChain, while conflicted is used to determine the output value of it. I know it doesn't matter here because you check for 0 or -1, but it's be much cleaner to call the INTERNAL version here, I think?